pylearner.ts

```diff
@@ -1350,9 +1350,169 @@ class ArrayTypeExpression extends TypeExpression {
+    const rvHasAlias = variables.includes(rv);
+    let newMayAliasSets : Immutable.Set<Immutable.Set<string>> =  ImmutableSet();
+    if (!lvHasAlias&&!rvHasAlias) {
+      let newMayAliasSet =  Immutable.Set.of(lv, rv);
+      newMayAliasSets = this.mayAliasSets.add(newMayAliasSet);
+    } else if (!lvHasAlias&&rvHasAlias) {
+      this.mayAliasSets.forEach(s => {
+        if (s.includes(rv))
+          s = s.add(lv);
+        newMayAliasSets = newMayAliasSets.add(s);
+      });
+    } else if (lvHasAlias&&!rvHasAlias) {
+      this.mayAliasSets.forEach(s => {
+        if ( s.includes(lv))
+          s = s.delete(lv);
+        newMayAliasSets = newMayAliasSets.add(s);
+      });
+      const newMayAliasSet = ImmutableSet.of(lv, rv);
+      newMayAliasSets = newMayAliasSets.add(newMayAliasSet);
+    } else {
+      this.mayAliasSets.forEach(s => {
+        let newMayAliasSet = s;
+        if (s.includes(lv)&&!s.includes(rv))
+          newMayAliasSet = newMayAliasSet.delete(lv);
+        if (s.includes(rv)&&!s.includes(lv))
+          newMayAliasSet = newMayAliasSet.add(lv);
+        newMayAliasSets = newMayAliasSets.add(newMayAliasSet);
+      });
+    }
+    const newMayAliasRelation = new MayAliasRelation(newMayAliasSets);
+    return newMayAliasRelation;
+  }
+
+  /**
+   * Returns a relation that relates variables x and y if and only if the relation holds for x and y
+   * 
+   * @returns relation
+   */
+  private removeSubsets(): Immutable.Set<Immutable.Set<string>> {
+    let mayAliasSetsWithoutSubsets: Immutable.Set<Immutable.Set<string>> =  ImmutableSet();
+    let mayAliasSetsToCheck = this.mayAliasSets;
+    for (var mayAliasSetToCheck of mayAliasSetsToCheck) {
+      let mayAliasSetsToDelete: Immutable.Set<Immutable.Set<string>> =  ImmutableSet();
+      let mayAliasSetToCheckIsSubsetOfAtLeastOne = false;
+      for (var mayAliasSetChecked of mayAliasSetsWithoutSubsets) {
+        const mayAliasSetToCheckIsSubset =  mayAliasSetToCheck.isSubset(mayAliasSetChecked);
+        const checkedMayAliasSetIsSubset = mayAliasSetChecked.isSubset(mayAliasSetToCheck);
+        if (checkedMayAliasSetIsSubset)
+          mayAliasSetsToDelete = mayAliasSetsToDelete.add(mayAliasSetChecked);
+        else if (mayAliasSetToCheckIsSubset) 
+          mayAliasSetToCheckIsSubsetOfAtLeastOne = true;
+      }
+      if (!mayAliasSetToCheckIsSubsetOfAtLeastOne) {
+        mayAliasSetsWithoutSubsets = mayAliasSetsWithoutSubsets.filter(s => !mayAliasSetsToDelete.includes(s));
+        mayAliasSetsWithoutSubsets = mayAliasSetsWithoutSubsets.add(mayAliasSetToCheck);
+      }
+    }
+    return mayAliasSetsWithoutSubsets;
+  }
+
+  /**
+   * Returns a relation that combines the relation of this with relation of o.
+   * 
+   * @param o relation to combine with
+   * @returns combined relation
+   */
+  combineWithMayAliasRelation(o: MayAliasRelation): MayAliasRelation {
+    let combinedMayAliasSets = this.mayAliasSets.concat(o.mayAliasSets);
```

Comment:
```suggestion
    let combinedMayAliasSets = this.mayAliasSets.union(o.mayAliasSets);
```
(`concat` en `union` doen precies hetzelfde maar `union` lijkt een logischere term als het om sets gaat.)